votes_up.php

```diff
@@ -188,23 +191,42 @@ function do_post_vote_up($post_id, $user_id, $vote)
         $forum_id = $post->post_parent;
         $forum = get_post_meta($forum_id);
     }
-    $pay_vote = $forum['_pay_vote'][0];
-    $pay_vote = $pay_vote;
 
+    $pay_vote = $forum['_pay_vote'][0];
     if ($pay_vote == 0) {
         return;
     } else if ($pay_vote == 1) {
         $cost_vote = $forum['_cost_vote'][0];
     }
-    //insert new vote
-    if ($voteplus) {
-        // Add points and save the current year as ref_id
-        mycred_add('vote_up', $author_id, $cost_vote, 'Vote_up', date('y'));
+
+    $post_meta = get_post_meta($post_id);
+    if (array_key_exists('_toggle_vote', $post_meta)) {
+        //insert new vote
+        if ($voteplus) {
+            // Add points and save the current year as ref_id
+            mycred_add('vote_up', $author_id, 2 * $cost_vote, 'Vote_up_change_vote', date('y'));
```

Comment:
I thought we discussed that previous vote should be deleted.. it would make mycred history much easier to read